### PR TITLE
[CHA-467] Add queries arguments

### PR DESCRIPTION
**Pull request by AdelELHAIBA-Akeneo**

### CODE REVIEW

File: content/graphql/common-notions.md [hunk 1]
```diff
@@ -267,5 +194,66 @@ To alias a field you only need to prefix the field with `your-desired-alias: `,
 }
 ```
 
-::: panel-link And now, let's discover how to resolve some use cases with GraphQL [Next](/graphql/use-cases.html)
-:::
+## Query with search
+
+**AS GraphQL use the Rest API**, all available filters can be used in the **Graphql queries** `search` **argument**
+
+You can a look at [all available filters](https://api.akeneo.com/documentation/filter.html)
```
Comment (tomglvng) [inline content/graphql/common-notions.md:201]:
```suggestion
You can have a look at [all available filters](https://api.akeneo.com/documentation/filter.html)
```

**Review by JulienVerbrugge — APPROVED** [review]

File: content/graphql/common-notions.md [hunk 2]
```diff
@@ -267,5 +194,66 @@ To alias a field you only need to prefix the field with `your-desired-alias: `,
...
+
+Some examples:
+- `{"updated":[{"operator":"SINCE LAST N DAYS","value":4}]}`
+- `{"created":[{"operator":"=","value":"2016-07-04 10:00:00"}]}`
+
+This filters need to be `json escaped` before being able to be usable inside GraphQL.
```
Comment (tomglvng) [inline content/graphql/common-notions.md:207]:
```suggestion
These filters need to be `JSON-escaped` before being able to be usable inside GraphQL.
```

File: content/graphql/common-notions.md [hunk 3]
```diff
@@ -267,5 +194,66 @@ To alias a field you only need to prefix the field with `your-desired-alias: `,
...
+
+You can use **your preferred online json escaper**.
```
Comment (tomglvng) [inline content/graphql/common-notions.md:209]:
```suggestion
You can use **your preferred online JSON escaper**.
```

File: content/graphql/queries-and-arguments.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
+This page present **all available arguments** for **the different queries**.
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:1]:
```suggestion
This page presents **all available arguments** for **the different queries**.
```

File: content/graphql/queries-and-arguments.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+## products
+Get a list of products (simple/variants).
+
+Based on the REST API endpoint [https://api.akeneo.com/api-reference.html#get_products_uuid](https://api.akeneo.com/api-reference.html#get_products_uuid)
+
+The following arguments are available:
+
+| Argument           | Type         | Description                                                                                                                                                                           |
+|--------------------|--------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `limit`            | **Int**      | Limit the number of result<br/>example `limit: 5`                                                                                                                                     |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:12]:
```suggestion
| `limit`            | **Int**      | Limit the number of results<br/>example `limit: 5`                                                                                                                                     |
```

File: content/graphql/queries-and-arguments.md [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+| `page`             | **String**   | Display a specific page, the value must come from the links{next/self/first}<br/>more details on [Common notions - paginations](/graphql/common-notions.html#working-with-pagination) |
+| `locales`          | **String[]** | Filter values to keep only the one related to these locales<br/>example `locales: ["fr_FR", "en_US"]`                                                                                 |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:14]:
```suggestion
| `locales`          | **String[]** | Filter values to keep only the ones related to these locales<br/>example `locales: ["fr_FR", "en_US"]`                                                                                 |
```

File: content/graphql/queries-and-arguments.md [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+| `channel`          | **String**   | Filter values to keep only the one related to this channel<br/>example `channel: "b2b"`                                                                                               |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:15]:
```suggestion
| `channel`          | **String**   | Filter values to keep only the ones related to this channel<br/>example `channel: "b2b"`                                                                                               |
```

File: content/graphql/queries-and-arguments.md [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+| `currencies`       | **String[]** | Filter values to keep only the one related to these currencies<br/>example `currencies: ["USD", "EUR"]`                                                                               |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:16]:
```suggestion
| `currencies`       | **String[]** | Filter values to keep only the ones related to these currencies<br/>example `currencies: ["USD", "EUR"]`                                                                               |
```

File: content/graphql/queries-and-arguments.md [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+| `search`           | **JSON**     | Filter results based on specific search<br/>more details on [Common notions - query with search](/graphql/common-notions.html#query-with-search)                                      |
+| `categories`       | **String[]** | Get product associated with these categories based on categories codes<br/>example `categories: ["categ-code1", "categ-code2"]`                                                       |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:18]:
```suggestion
| `categories`       | **String[]** | Get products associated with these categories based on categories codes<br/>example `categories: ["categ-code1", "categ-code2"]`                                                       |
```

File: content/graphql/queries-and-arguments.md [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+| `families`         | **String[]** | Get product associated with these families based on families codes<br/>example `families: ["family-code1", "family-code2"]`                                                           |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:19]:
```suggestion
| `families`         | **String[]** | Get products associated with these families based on family codes<br/>example `families: ["family-code1", "family-code2"]`                                                           |
```

File: content/graphql/queries-and-arguments.md [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+:::
+
+## productModels
+Get a list of product models. 
+
+Based on the REST API endpoint [https://api.akeneo.com/api-reference.html?#Productmodel](https://api.akeneo.com/api-reference.html?#Productmodel)
+
+| Argument           | Type         | Description                                                                                                                                                                               |
+|--------------------|--------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `limit`            | **Int**      | Limit the number of result<br/>example `limit: 5`                                                                                                                                         |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:38]:
```suggestion
| `limit`            | **Int**      | Limit the number of results<br/>example `limit: 5`                                                                                                                                         |
```

File: content/graphql/queries-and-arguments.md [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+| `page`             | **String**   | Display a specific page, the value must come from the **links{next/self/first}**<br/>more details on [Common notions - paginations](/graphql/common-notions.html#working-with-pagination) |
+| `channel`          | **String**   | Filter values to keep only the one related to this channel<br/>example `channel: "b2b"`                                                                                                   |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:40]:
```suggestion
| `channel`          | **String**   | Filter values to keep only the ones related to this channel<br/>example `channel: "b2b"`                                                                                                   |
```

File: content/graphql/queries-and-arguments.md [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+| `locales`          | **String[]** | Filter values to keep only the one related to this locales<br/>example `locales: ["fr_FR", "en_US"]`                                                                                      |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:41]:
```suggestion
| `locales`          | **String[]** | Filter values to keep only the ones related to this locales<br/>example `locales: ["fr_FR", "en_US"]`                                                                                      |
```

File: content/graphql/queries-and-arguments.md [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+| `currencies`       | **String[]** | Filter values to keep only the one related to these currencies<br/>example `currencies: ["USD", "EUR"]`                                                                                   |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:42]:
```suggestion
| `currencies`       | **String[]** | Filter values to keep only the ones related to these currencies<br/>example `currencies: ["USD", "EUR"]`                                                                                   |
```

File: content/graphql/queries-and-arguments.md [hunk 15]
```diff
@@ -0,0 +1,223 @@
...
+| `search`           | **JSON**     | Filter results based on specific search<br/>more details on [Common notions - query with search](/graphql/common-notions.html#query-with-search)                                          |
+| `codes`            | **String[]** | Get only specific items based on their codes<br/>example `codes: ["code1", "code-2"]`                                                                                                     |
+| `categories`       | **String[]** | Get product associated with these categories based on categories codes<br/>example `categories: ["categ-code1", "categ-code2"]`                                                           |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:45]:
```suggestion
| `categories`       | **String[]** | Get products associated with these categories based on categories codes<br/>example `categories: ["categ-code1", "categ-code2"]`                                                           |
```

File: content/graphql/queries-and-arguments.md [hunk 16]
```diff
@@ -0,0 +1,223 @@
...
+| `families`         | **String[]** | Get product associated with these families based on families codes<br/>example `families: ["family-code1", "family-code2"]`                                                               |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:46]:
```suggestion
| `families`         | **String[]** | Get products associated with these families based on families codes<br/>example `families: ["family-code1", "family-code2"]`                                                               |
```

File: content/graphql/queries-and-arguments.md [hunk 17]
```diff
@@ -0,0 +1,223 @@
...
+:::
+
+## families
+Get a list of families. 
+
+Based on the REST API endpoint [https://api.akeneo.com/api-reference.html?#Family](https://api.akeneo.com/api-reference.html?#Family)
+
+| Argument  | Type         | Description                                                                                                                                                                           |
+|-----------|--------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `limit`   | **Int**      | Limit the number of result<br/>example `limit: 5`                                                                                                                                     |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:62]:
```suggestion
| `limit`   | **Int**      | Limit the number of results<br/>example `limit: 5`                                                                                                                                     |
```

File: content/graphql/queries-and-arguments.md [hunk 18]
```diff
@@ -0,0 +1,223 @@
...
+| `locales` | **String[]** | Filter values to keep only the one related to these locales<br/>example `locales: ["fr_FR", "en_US"]`                                                                                 |
+
+## categories
+Get a list of categories. 
+
+Based on the REST API endpoint [https://api.akeneo.com/api-reference.html?#Category](https://api.akeneo.com/api-reference.html?#Category)
+
+| Argument  | Type         | Description                                                                                                                                                                           |
+|-----------|--------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `limit`   | **Int**      | Limit the number of result<br/>example `limit: 5`                                                                                                                                     |
```
Comment (tomglvng) [inline content/graphql/queries-and-arguments.md:74]:
```suggestion
| `limit`   | **Int**      | Limit the number of results<br/>example `limit: 5`                                                                                                                                     |
```

**Comment by akeneo-secops** [comment]
### Gitleaks PR check
🟢 We detected no secrets within this PR! Good job 🎉

**Review by tomglvng — CHANGES_REQUESTED** [review]

**Pull request merged**